transaction-view/src/static_accounts_meta.rs

```diff
@@ -0,0 +1,95 @@
+use {
+    crate::{
+        bytes::{offset_array_len, read_byte},
+        result::{Result, TransactionParsingError},
+    },
+    solana_sdk::{packet::PACKET_DATA_SIZE, pubkey::Pubkey},
+};
+
+/// Contains meta-data about the static account keys in a transaction packet.
+#[derive(Default)]
+pub struct StaticAccountMeta {
+    /// The number of static accounts in the transaction.
+    pub(crate) num_static_accounts: u16,
+    /// The offset to the first static account in the transaction.
+    pub(crate) offset: u16,
+}
+
+impl StaticAccountMeta {
+    pub fn try_new(bytes: &[u8], offset: &mut usize) -> Result<Self> {
+        // The packet has a maximum length of 1232 bytes.
+        // This means the maximum number of 32 byte keys is 38.
+        // 38 as an min-sized encoded u16 is 1 byte.
+        // We can simply read this byte, if it's >38 we can return None.
+        const MAX_STATIC_ACCOUNTS_PER_PACKET: u16 =
+            (PACKET_DATA_SIZE / core::mem::size_of::<Pubkey>()) as u16;
```

Comment:
Is `StaticAccountMeta` going to represent `AccountMeta`?

https://github.com/anza-xyz/agave/blob/51208f8eebf1d5c3eba6d2d63de1b6673cd3d173/sdk/program/src/instruction.rs#L544-L551

I was wondering if you should account for the two booleans in addition to the public key.

Reply:
no unrelated to that struct. Have just been naming all my internal transaction view structs holding offsets/sizes with `Meta` suffixes.

The `is_signer` and `is_writable` are not part of the actual packet and are implied by combination of account index and information from the header.
So these will be functions on the wrapping `transaction_view` once I add the accessor functions.

Reply:
I mixed up the names and thought they were related. How about calling the struct `StaticAccountKeyMeta`?

Reply:
59cdc7c523177b989990821afe3f6572eff9ada0